addons/website/models/website_visitor.py

```diff
@@ -267,62 +312,49 @@ def _create_visitor(self):
         if tz:
             vals['timezone'] = tz
 
-        if not self.env.user._is_public():
-            vals['partner_id'] = self.env.user.partner_id.id
-            vals['name'] = self.env.user.partner_id.name
-        return self.sudo().create(vals)
+        # Add missing default values and convert to SQL format
+        vals = self._prepare_create_values([vals])[0]
+        vals = {k: self._fields[k].convert_to_column(v, self) for k, v in vals.items()}
```

Comment:
Please use meaningful variable names (and remove unnecessary variable assignment when possible), like
```
creation_vals = self._prepare_create_values([creation_vals])[0]
return {
    field_name: self._fields[field_name].convert_to_column(field_value, self)
    for field_name, field_value in creation_vals.items()
}
```